config/relay/registry.go

```diff
@@ -0,0 +1,92 @@
+package relay
+
+import "strings"
+
+// Registry keeps an in-memory registry of all known relays.
+//
+// It contains a mapping for all added validators and their corresponding relays.
+// It also has default fallback relays which are used when a provided validator has no relays.
+type Registry struct {
+	defaultRelays  Set
+	relaysByPubKey relaysByValidatorPubKey
+}
+
+// NewProposerRegistry creates a new instance of Registry.
+func NewProposerRegistry() *Registry {
+	return &Registry{
+		defaultRelays:  NewRelaySet(),
+		relaysByPubKey: newRelaysByValidatorPubKey(),
+	}
+}
+
+// AddRelayForProposer adds a relay entry for the given proposer.
+func (r *Registry) AddRelayForProposer(key ValidatorPublicKey, entry Entry) {
+	r.relaysByPubKey.Add(key, entry)
+}
+
+// AddEmptyProposer adds a proposer with an empty relay set.
+func (r *Registry) AddEmptyProposer(key ValidatorPublicKey) {
```

Comment:
non-blocking: should we use the naming `empty` or `disabled` when the MEV is disabled for that proposed? as empty could be confusing in this context.

Reply:
Agree